### DISCUSSION TITLE
I see the message that "disapproved requested chains"

### DISCUSSION THREAD

**Discussion by matthew2099** [WalletConnect/WalletConnectUnity#215] [Q&A]
I connected klaytn baobab.
But OkxWallet don't display network.
And i success connect and i try to transaction.
And i see the message "disapproved requested chains"
I wonder OkxWallet is don't able to testnet??

![image](https://github.com/WalletConnect/WalletConnectUnity/assets/109068040/a32f0804-c2f2-4313-bc76-3bbbcb28f48d)
![image](https://github.com/WalletConnect/WalletConnectUnity/assets/109068040/6ba65ac5-cc33-436a-b0d9-3047ce440776)

**Comment by skibitsky** [comment]
In the OKX Wallet, you can add a custom network from the wallet settings. Try adding test chain from there.

**Reply by matthew2099** [reply]
Already i add custom chain.
But don't display
I can personal sign but can't transaction

**Reply by skibitsky** [reply]
Do you have `eth_sendTransaction` in namespaces? Does it work with other chains?

**Reply by matthew2099** [reply]
I setting that.
And i don't see the work transaction

![image](https://github.com/WalletConnect/WalletConnectUnity/assets/109068040/2dd2349f-6152-4709-9a65-85a81b56ebdc)

**Reply by skibitsky** [reply]
Does the transaction UI appear in the wallet if you use chain ID `eip155:1` instead?

**Reply by matthew2099** [reply]
I try that.
But same message that disapproved requested chains.

![image](https://github.com/WalletConnect/WalletConnectUnity/assets/109068040/dcaa074c-5767-4e11-83b7-2d1a0c65fbdd)

**Reply by skibitsky** [reply]
Thank you. Could you please share the whole connection and transaction code? I'll try to reproduce it myself. Also, do you use OKX on iOS or Android?

**Reply by matthew2099** [reply]
private const string COINABI =
            "[   {     \"inputs\": [       {         \"internalType\": \"string\",         \"name\": \"name\",         \"type\": \"string\"       },       {         \"internalType\": \"string\",         \"name\": \"symbol\",         \"type\": \"string\"       },       {         \"internalType\": \"address\",         \"name\": \"to\",         \"type\": \"address\"       },       {         \"internalType\": \"uint256\",         \"name\": \"initialSupply\",         \"type\": \"uint256\"       }     ],     \"stateMutability\": \"nonpayable\",     \"type\": \"constructor\"   },   {     \"anonymous\": false,     \"inputs\": [       {         \"indexed\": true,         \"internalType\": \"address\",         \"name\": \"owner\",         \"type\": \"address\"       },       {         \"indexed\": true,         \"internalType\": \"address\",         \"name\": \"spender\",         \"type\": \"address\"       },       {         \"indexed\": false,         \"internalType\": \"uint256\",         \"name\": \"value\",         \"type\": \"uint256\"       }     ],     \"name\": \"Approval\",     \"type\": \"event\"   },   {     \"anonymous\": false,     \"inputs\": [       {         \"indexed\": true,         \"internalType\": \"address\",         \"name\": \"from\",         \"type\": \"address\"       },       {         \"indexed\": true,         \"internalType\": \"address\",         \"name\": \"to\",         \"type\": \"address\"       },       {         \"indexed\": false,         \"internalType\": \"uint256\",         \"name\": \"value\",         \"type\": \"uint256\"       }     ],     \"name\": \"Transfer\",     \"type\": \"event\"   },   {     \"inputs\": [       {         \"internalType\": \"address\",         \"name\": \"owner\",         \"type\": \"address\"       },       {         \"internalType\": \"address\",         \"name\": \"spender\",         \"type\": \"address\"       }     ],     \"name\": \"allowance\",     \"outputs\": [       {         \"internalType\": \"uint256\",         \"name\": \"\",         \"type\": \"uint256\"       }     ],     \"stateMutability\": \"view\",     \"type\": \"function\"   },   {     \"inputs\": [       {         \"internalType\": \"address\",         \"name\": \"spender\",         \"type\": \"address\"       },       {         \"internalType\": \"uint256\",         \"name\": \"amount\",         \"type\": \"uint256\"       }     ],     \"name\": \"approve\",     \"outputs\": [       {         \"internalType\": \"bool\",         \"name\": \"\",         \"type\": \"bool\"       }     ],     \"stateMutability\": \"nonpayable\",     \"type\": \"function\"   },   {     \"inputs\": [       {         \"internalType\": \"address\",         \"name\": \"account\",         \"type\": \"address\"       }     ],     \"name\": \"balanceOf\",     \"outputs\": [       {         \"internalType\": \"uint256\",         \"name\": \"\",         \"type\": \"uint256\"       }     ],     \"stateMutability\": \"view\",     \"type\": \"function\"   },   {     \"inputs\": [       {         \"internalType\": \"uint256\",         \"name\": \"amount\",         \"type\": \"uint256\"       }     ],     \"name\": \"burn\",     \"outputs\": [],     \"stateMutability\": \"nonpayable\",     \"type\": \"function\"   },   {     \"inputs\": [       {         \"internalType\": \"address\",         \"name\": \"account\",         \"type\": \"address\"       },       {         \"internalType\": \"uint256\",         \"name\": \"amount\",         \"type\": \"uint256\"       }     ],     \"name\": \"burnFrom\",     \"outputs\": [],     \"stateMutability\": \"nonpayable\",     \"type\": \"function\"   },   {     \"inputs\": [],     \"name\": \"decimals\",     \"outputs\": [       {         \"internalType\": \"uint8\",         \"name\": \"\",         \"type\": \"uint8\"       }     ],     \"stateMutability\": \"view\",     \"type\": \"function\"   },   {     \"inputs\": [       {         \"internalType\": \"address\",         \"name\": \"spender\",         \"type\": \"address\"       },       {         \"internalType\": \"uint256\",         \"name\": \"subtractedValue\",         \"type\": \"uint256\"       }     ],     \"name\": \"decreaseAllowance\",     \"outputs\": [       {         \"internalType\": \"bool\",         \"name\": \"\",         \"type\": \"bool\"       }     ],     \"stateMutability\": \"nonpayable\",     \"type\": \"function\"   },   {     \"inputs\": [       {         \"internalType\": \"address\",         \"name\": \"spender\",         \"type\": \"address\"       },       {         \"internalType\": \"uint256\",         \"name\": \"addedValue\",         \"type\": \"uint256\"       }     ],     \"name\": \"increaseAllowance\",     \"outputs\": [       {         \"internalType\": \"bool\",         \"name\": \"\",         \"type\": \"bool\"       }     ],     \"stateMutability\": \"nonpayable\",     \"type\": \"function\"   },   {     \"inputs\": [],     \"name\": \"name\",     \"outputs\": [       {         \"internalType\": \"string\",         \"name\": \"\",         \"type\": \"string\"       }     ],     \"stateMutability\": \"view\",     \"type\": \"function\"   },   {     \"inputs\": [],     \"name\": \"symbol\",     \"outputs\": [       {         \"internalType\": \"string\",         \"name\": \"\",         \"type\": \"string\"       }     ],     \"stateMutability\": \"view\",     \"type\": \"function\"   },   {     \"inputs\": [],     \"name\": \"totalSupply\",     \"outputs\": [       {         \"internalType\": \"uint256\",         \"name\": \"\",         \"type\": \"uint256\"       }     ],     \"stateMutability\": \"view\",     \"type\": \"function\"   },   {     \"inputs\": [       {         \"internalType\": \"address\",         \"name\": \"to\",         \"type\": \"address\"       },       {         \"internalType\": \"uint256\",         \"name\": \"amount\",         \"type\": \"uint256\"       }     ],     \"name\": \"transfer\",     \"outputs\": [       {         \"internalType\": \"bool\",         \"name\": \"\",         \"type\": \"bool\"       }     ],     \"stateMutability\": \"nonpayable\",     \"type\": \"function\"   },   {     \"inputs\": [       {         \"internalType\": \"address\",         \"name\": \"from\",         \"type\": \"address\"       },       {         \"internalType\": \"address\",         \"name\": \"to\",         \"type\": \"address\"       },       {         \"internalType\": \"uint256\",         \"name\": \"amount\",         \"type\": \"uint256\"       }     ],     \"name\": \"transferFrom\",     \"outputs\": [       {         \"internalType\": \"bool\",         \"name\": \"\",         \"type\": \"bool\"       }     ],     \"stateMutability\": \"nonpayable\",     \"type\": \"function\"   } ]";
            
   var args = JsonConvert.SerializeObject(new string[]
        {
            "0x47dA1DD8D7E1ec78F30eaD1a32F2B7047A018F04", "0"
        });
        var data = await CreateContractData(COINABI, "transfer", args);

        var request = new EthSendTransaction(new Transaction
        {
            to = "0xdaaa0048f22d75f008916614545aab7845122e10",
            value = "0",
            data = data,
        });

        try
        {
            var result = await WalletConnect.Instance.RequestAsync<EthSendTransaction, string>(request);
            
            text.text = "Transaction success";
        }
        catch (WalletConnectException e)
        {
            //Debug.Log($"[WalletConnectModalSample] Transaction Error: {e.Message}");
            text.text = $"[WalletConnectModalSample] Transaction Error: {e.Message}";
        }

This is my code. I use android OKX.
And i know our network is klaytn baobab

And data is 0xa9059cbb00000000000000000000000047da1dd8d7e1ec78f30ead1a32f2b7047a018f040000000000000000000000000000000000000000000000000000000000000000

**Reply by skibitsky** [reply]
What is the code for `CreateContractData` method?

**Reply by matthew2099** [reply]
Ah. other sdk method.
Actually i don't understand Nethereum ABI Encoding.
So i use that.
Hum...maybe this is message reason?? i don't know that...

**Reply by skibitsky** [reply]
I think it may be a data encoding issue. You can ask Nethereum-related questions in their Discord. The library author is quite active there: [Nethereum Discord](https://discord.gg/u3Ej2BReNn)

**Reply by matthew2099** [reply]
oh i see. i try that

**Reply by matthew2099** [reply]
I failed that. because i have a plugin conflict.
and that method is use metamask wallet and i check normal work.
I think okx wallet is strange.

**Reply by matthew2099** [reply]
and i find OptionalMethods. hum can i test that??

**Reply by matthew2099** [reply]
My mistake.
I write from and gas. so work
